mlir/include/mlir/Dialect/Vector/Transforms/VectorRewritePatterns.h

```diff
@@ -389,6 +389,12 @@ void populateVectorLinearizeTypeConversionsAndLegality(
     TypeConverter &typeConverter, RewritePatternSet &patterns,
     ConversionTarget &target, unsigned targetBitWidth);
 
+/// Populates patterns for linearizing ND (N >= 2) vector operations to 1D
+/// vector shuffle operations.
+void populateVectorLinearizeToShuffleRewritePatterns(
```

Comment:
Extract ops are "shuffle-like" ops so perhaps we can call this `populateVectorLinearizeShuffleLikeOpsPatterns` or similar

Reply:
renamed to `populateVectorLinearizeShuffleLikeOpsPatterns`